common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1488,13 +1488,13 @@ public UTF8String[] split(UTF8String pattern, int limit) {
       int newLimit = limit > numChars() || limit <= 0 ? numChars() : limit;
       byte[] input = getBytes();
       int byteIndex = 0;
-      int charIndex = 0;
       UTF8String[] result = new UTF8String[newLimit];
-      while (charIndex < newLimit) {
+      for (int i = 0; i < newLimit - 1; i++) {
         int currCharNumBytes = numBytesForFirstByte(input[byteIndex]);
-        result[charIndex++] = UTF8String.fromBytes(input, byteIndex, currCharNumBytes);
+        result[i] = UTF8String.fromBytes(input, byteIndex, currCharNumBytes);
         byteIndex += currCharNumBytes;
       }
+      result[newLimit - 1] = UTF8String.fromBytes(input, byteIndex, numBytes() - byteIndex);
```

Comment:
is `ArrayIndexOutOfBoundsException` possible here?
what if newLimit=0 (i.e. numChars()=0, limit=-1)

Reply:
> is `ArrayIndexOutOfBoundsException` possible here? what if newLimit=0 (i.e. numChars()=0, limit=-1)

no, this code block will only be entered when the following conditions are met. 
```
if (numBytes() != 0 && pattern.numBytes() == 0)
```